controllers/operands/networkAddons.go

```diff
@@ -113,28 +119,40 @@ func (*cnaHooks) updateSpec(req *common.HcoRequest, found *networkaddonsv1.Netwo
 	return false
 }
 
-// If deployOVS annotation doesn't exists prior the upgrade - set this annotation to true;
+// If annotation doesn't exists prior the upgrade - set this annotation to true;
 // Otherwise - remain the value as it is.
-func setDeployOvsAnnotation(req *common.HcoRequest, found *networkaddonsv1.NetworkAddonsConfig) {
+func setAnnotation(req *common.HcoRequest, found *networkaddonsv1.NetworkAddonsConfig, annotation string) {
 	if req.UpgradeMode {
-		_, exists := req.Instance.Annotations["deployOVS"]
+		_, exists := req.Instance.Annotations[annotation]
 		if !exists {
 			if req.Instance.Annotations == nil {
 				req.Instance.Annotations = map[string]string{}
 			}
-			if found.Spec.Ovs != nil {
-				req.Instance.Annotations["deployOVS"] = "true"
-				req.Logger.Info("deployOVS annotation is set to true.")
+			if checkComponent(found.Spec, annotation) {
+				req.Instance.Annotations[annotation] = "true"
+				req.Logger.Info(annotation + " annotation is set to true.")
 			} else {
-				req.Instance.Annotations["deployOVS"] = "false"
-				req.Logger.Info("deployOVS annotation is set to false.")
+				req.Instance.Annotations[annotation] = "false"
+				req.Logger.Info(annotation + " annotation is set to false.")
 			}
 
 			req.Dirty = true
 		}
 	}
 }
 
+func checkComponent(spec networkaddonsshared.NetworkAddonsConfigSpec, annotation string) bool {
+	if annotation == deployOVS {
+		return spec.Ovs != nil
+	}
+	if annotation == deployKSD {
+		return spec.KubeSecondaryDNS != nil
+	}
+
+	logger.Error(errors.New("checkComponent error"), "unknown annotation")
+	return false
```

Comment:
maybe panic here instead?

Reply:
I think that panic from a server should be done for a very good reason. I'm not sure this specific case. Anyway, if you decide to change it, please add a detailed comment.